inc/ticketcontract.class.php

```diff
@@ -0,0 +1,192 @@
+<?php
+/**
+ * ---------------------------------------------------------------------
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2015-2021 Teclib' and contributors.
+ *
+ * http://glpi-project.org
+ *
+ * based on GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2003-2014 by the INDEPNET Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+if (!defined('GLPI_ROOT')) {
+   die("Sorry. You can't access this file directly");
+}
+
+class TicketContract extends CommonItilObject_Item {
+
+   public static $itemtype_1 = 'Ticket';
+   public static $items_id_1 = 'tickets_id';
+
+   public static $itemtype_2 = 'Contract';
+   public static $items_id_2 = 'contracts_id';
+   public static $checkItem_2_Rights = self::HAVE_VIEW_RIGHT_ON_ITEM;
+
+   public static function getTypeName($nb = 0) {
+      return _n('fsefesfsfe', 'fsefsefsef', $nb);
```

Comment:
`fsefesfsfe` ? :)

Reply:
Yeah maybe we should not keep that :D 
To my defense I don't think it is actually displayed anywhere so it didn't catch it when testing the feature.